### ISSUE TITLE
Fish does not work in a TTY session.

### ISSUE THREAD

**Issue by dark-yux** [fish-shell/fish-shell#2070]
In ubuntu if you set fish as your default shell and press ctrl+alt+f1 log in and use fish, the history colors etc are all broken.

**Comment by hai-ld** [comment]
It works for me (fish 2.1.1, Ubuntu 14.04.2). Could you elaborate the issue?

**Comment by dark-yux** [comment]
Mabie it was the colors I was using the current version works fine It just prints weird characters before the prompt.

**Comment by zanchey** [comment]
My guess is that you are printing UTF-8 to a non-capable terminal.

Can you confirm the version you are running, the version of Ubuntu you are using and the value of `echo $TERM` when you are logged in on the console?

**Comment by dark-yux** [comment]
`echo $TERM` outputs "linux", `echo $FISH_VERSION` outputs "2.1.1", and `cat /etc/issue` outputs "Ubuntu 15.04"

**Comment by zanchey** [comment]
I also meant to ask for the output of `locale`.

**Comment by zanchey** [comment]
Basically, I'm wondering if you're seeing this - http://superuser.com/questions/556993/how-to-display-unicode-in-a-linux-virtual-terminal

**Comment by dark-yux** [comment]
I'm sorry I switched linuxes I'm now using Sabayon and have not checked for the same problems but I believe my locale was probably en_UTF8.

**Comment by pickfire** [comment]
I have this problem too, but after I switched to the **default** theme, it works fine. I guess it is the problem of character support in framebuffer, since it doesn't work with some arrows.

**Comment by zanchey** [comment]
I don't think there's much we can do about this. AFAICT, the text-mode console only supports 16 colours and 512 glyphs, and even with fbdev supporting more colors the console fonts don't implement all of Unicode anyway. KMSCON is probably the way to go if you need arrows or many fancy colors on your terminal.

**Comment by pickfire** [comment]
What about disabling arrows in framebuffer by changing the fish theme?

```fish
if test $SSH_tty
    printf '% '
    return 0
end
```

**Comment by Lucretiel** [comment]
Wouldn't [isatty](http://fishshell.com/docs/current/commands.html#isatty) be more idomatic?

**Comment by pickfire** [comment]
Sorry I didn't meant to post that here. What I mean is:
```fish
# If login shell, don't do anything
if status --is-login; and function fish_prompt; echo -n '% '; end; end
```

**Comment by faho** [comment]
> Note: I uses echo because printf doesn't work

printf needs two "%"'s to print a literal "%", i.e:

    if $TERM = "linux"; function fish_prompt; printf '%%'; end; end

(the and is unnecessary since "if" has not only a condition but also a body, and the "$TERM = linux" is the more accurate check - an xterm can also run a login shell)

Really, I don't think this is going to improve much - the in-kernel VT sucks, which is why things like "kmscon" and the spun-off "systemd-consoled" exist.

What we can do is add a fallback prompt to all sample prompts that kicks in when "$TERM = linux" like I did above (note: this should be _after_ the normal prompt definition or you'd need to put that in an else-block), and we can also add fallback-colors since VTs aren't going to support 256 colors.

Or we can recommend people use a userspace-vt (like kmscon) and actually have a nice experience.

**Comment by faho** [comment]
Since I'm not quite sure GH notifies for references, I'm gonna say it here again:

I'm okay with #2311 - I've written it after all. If any of you isn't, please speak up and we'll try to find an alternative solution.

**Comment by faho** [comment]
Fixed by f71e877f18037f71c00f81e30c1173f51c42a58c.
